changes/4093-timkpaine.md

```diff
@@ -0,0 +1,3 @@
+Allow for shallow copies of attributes, adjusting the behavior of #3642
+`Config.copy_on_model_validation` is now a str enum of `["none", "deep", "shallow"]` corresponding to 
+not copying, deep copy, shallow copy.
```

Comment:
My main concern is on the default value

With the test example
```py
class User(BaseModel):
    hobbies: list[str]

my_user = User(hobbies=['scuba diving'])

class Transaction(BaseModel):
    user: User

t = Transaction(user=my_user)

print(t.user is my_user)
print(t.user.hobbies is my_user.hobbies)
```

we used to have in v1.9.0
```
copy_on_model_validation = True (default): False, True
copy_on_model_validation = False: True, True
```

which got changed (the famous breaking change) in 1.9.1

```
copy_on_model_validation = True (default): False, False <-💥
copy_on_model_validation = False: True, True
```

We now have with this patch
```
copy_on_model_validation = 'deep' : False, False
copy_on_model_validation = 'shallow': False, True
copy_on_model_validation = 'none': True, True
```

I wonder if we should write a note saying a breaking change has been introduced by mistake in a release version and that the old 1.9.0 behaviour has been fixed. In which case, the default value should be `'shallow'`.

Reply:
I agree, we should make the default `shallow`.

I'll try to fix this today.

Reply:
default changed.